### DISCUSSION TITLE
CSS.supports undefined

### DISCUSSION THREAD

**Discussion by WebMechanic** [capricorn86/happy-dom#1166] [Q&A] [answered]
Hello,
I'm a bit confused...

It seems `CSS.ts` implements the CSS interface, however ...
the snippet below is some bit from a Vitest (0.34.6) test case with "@vitest-environment happy-dom" and `@happy-dom/global-registrator` set, loaded and enabled.

I need to test a script that makes calls to `CSS.supports()` which fails with "supports is not a function". I'm **not interested in any results of CSS.supports()** but the test suite won't execute load bc. of "supports" miraculously being undefined.

```js
/**
 * @vitest-environment happy-dom
 */
test('CSS.supports', (ctx) => {
  expect(window).toBeDefined();       // pass
  expect(document).toBeDefined();     // pass
  expect(window.CSS).toBeDefined();   // pass
  expect(CSS).toBeDefined();          // pass

// throws "expected undefined not to be undefined"
  expect(CSS.supports).toBeDefined();
// throws "expected false to be truthy"
  expect('supports' in window.CSS).toBeTruthy();  
})
```

Why does `window.CSS` pass but `CSS.supports` is reported to be "undefined" and `CSS.supports("fubar")` fails with "supports is not a function"?

What am I missing here? 
The script to be tested runs "flawless" in a real browser but has no test coverage because of this weirness.

Thanks 4 help.

**Comment by capricorn86** [comment, marked as answer]
Sorry for answering this so late.

I believe this was a Vitest bug and was fixed by https://github.com/vitest-dev/vitest/pull/4601
